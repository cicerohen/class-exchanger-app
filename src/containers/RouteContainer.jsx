import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import { Route as ReactRoute, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import { auth, router } from '../redux';

class RouteContainer extends Component {
  render() {
    const {
      isProtected, isAuthenticating, isAuthenticated,
      location, previousRoute, ...others
    } = this.props;
    if (isAuthenticating) {
      return (<div>Loading</div>);
    }
    if (isAuthenticated && (location.pathname === '/signin' || location.pathname === '/signup')) {
      return (<Redirect to={previousRoute.get('pathname')} />);
    }
    if (!isAuthenticated && isProtected) {
      return (<Redirect to="/signin" />);
    }
    return (<ReactRoute {...others} />);
  }
}

RouteContainer.defaultProps = {
  isProtected: false,
  isAuthenticated: false,
  isAuthenticating: false,
};

RouteContainer.propTypes = {
  isProtected: PropTypes.bool,
  isAuthenticated: PropTypes.bool,
  isAuthenticating: PropTypes.bool,
  location: PropTypes.shape({
    pathname: PropTypes.any.isRequired,
  }).isRequired,
  previousRoute: PropTypes.objectOf(PropTypes.any).isRequired,
};

const mapStateToProps = createStructuredSelector({
  isAuthenticating: auth.selectors.isAuthenticating,
  isAuthenticated: auth.selectors.isAuthenticated,
  previousRoute: router.selectors.getPreviousRoute,
});

export default connect(mapStateToProps)(RouteContainer);
