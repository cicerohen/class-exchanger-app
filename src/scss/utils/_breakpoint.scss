@function lower-bound($range) {
  @if length($range) <= 0 {
    @return 0;
  }
  @return nth($range,1);
}

@function upper-bound($range) {
  @if length($range) < 2 {
    @return 999999999999;
  }
  @return nth($range, 2);
}

$small-range:(0em, 39.9375em);
$medium-range:(40em, 63.9375em);
$large-range:(64em, 74.9375em);

$screen: "only screen" !default;
$small-only: "#{$screen} and (max-width: #{upper-bound($small-range)})" !default;
$medium-only: "#{$screen} and (min-width: #{lower-bound($medium-range)}) and (max-width: #{upper-bound($medium-range)})" !default;
$medium-up: "#{$screen} and (min-width: #{lower-bound($medium-range)})" !default;
$large-only: "#{$screen} and (min-width: #{lower-bound($large-range)}) and (max-width: #{upper-bound($large-range)})" !default;
$large-up: "#{$screen} and (min-width: #{lower-bound($large-range)})" !default;
