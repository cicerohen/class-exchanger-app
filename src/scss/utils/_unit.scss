@function strip-unit($num) {
  @return $num / ($num * 0 + 1);
}

@function rem-calc($value, $base: null) {
    
  @if $base == null {
    $base: 62.5%;
  }

  @if unit($base) == '%' {
    $base: ($base / 100%) * 16px;
  }

  @if type-of($value) != 'number' {
    @warn inspect($value) + ' was passed to rem-calc(), which is not a number.';
    @return $value;
  }

  @if unit($value) != 'rem' {
    $value: strip-unit($value) / strip-unit($base) * 1rem;
  }

  @if $value == 0rem {
    $value: 0;
  }
  
  @return $value;

}
