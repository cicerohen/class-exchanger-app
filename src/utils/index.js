import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));
export const emptyMap = asImmutable({});

export const buildActions = (moduleName, prefix, suffix) => ({
    REQUESTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}REQUESTED`,
    STARTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}STARTED`,
    SUCCEEDED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}SUCCEEDED`,
    FAILED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FAILED`,
    FULFILLED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FULFILLED`,
});

export const toArrayPath = path => path.split('/').filter(p => p.replace(' ', '').length > 0);

export const getHistory = () => {
    window.browserHistory = window.browserHistory || createBrowserHistory();
    return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;
  return nodeEnv !== 'production' && nodeEnv !== 'test';
};
