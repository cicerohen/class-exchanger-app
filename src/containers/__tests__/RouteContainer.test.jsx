import React from 'react';
import { shallow } from 'enzyme';
import { createReduxTestStore } from '../../utils/test';
import RouteContainer from '../RouteContainer';

const initialState = {
  router: {
    current: {
      pathname: '/',
    },
    previous: {
      pathname: '/',
    },
  },
};

const minimalProps = {
  location: {
    pathname: '/',
  },
};

describe('<RouteContainer /> Tests', () => {
  it('should render without exploding', (done) => {
    const testStore = createReduxTestStore({
      configs: { initialState },
    });
    const wrapper = shallow(<RouteContainer store={testStore} {...minimalProps} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly with minimal props', (done) => {
    const testStore = createReduxTestStore({
      configs: { initialState },
    });
    const wrapper = shallow(<RouteContainer store={testStore} {...minimalProps} />);
    expect(wrapper.dive().shallow()).toMatchSnapshot();
    done();
  });

  it('should render correctly when is not authenticated and route is protected', (done) => {
    const testStore = createReduxTestStore({
      configs: {
        initialState: {
          ...initialState,
          auth: {
            isAuthenticating: false,
          },
        },
      },
    });
    const wrapper = shallow(<RouteContainer store={testStore} {...minimalProps} isProtected />);
    expect(wrapper.dive()).toMatchSnapshot();
    done();
  });
});
