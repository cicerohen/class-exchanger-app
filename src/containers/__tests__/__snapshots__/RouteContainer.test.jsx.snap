// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteContainer /> Tests should render correctly when is not authenticated and route is protected 1`] = `
<Redirect
  push={false}
  to="/signin"
/>
`;

exports[`<RouteContainer /> Tests should render correctly with minimal props 1`] = `
<div>
  Loading
</div>
`;
